tcp_endpoint/src/tcp_endpoint/UnityTCPSender.py

```diff
@@ -0,0 +1,40 @@
+import rospy
+import socket
+from tcp_endpoint.RosTCPClientThread import ClientThread
+from tcp_endpoint.msg import RosUnityError
+
+class UnityTCPSender:
+    """
+    Connects and sends messages to the server on the Unity side.
+    """
+    def __init__(self, unity_ip, unity_port):
+        self.unity_ip = unity_ip
+        self.unity_port = unity_port
+        self.ip_is_overridden = (self.unity_ip != '')
+
+    def process_handshake(self, ip, port):

```

Comment:
Can you add a comment describing why unity_ip  and unity_port would need to be updated?